Description: Protect temporary filenames
 This patch protects temporary filenames created by
 sshuttle by chmodding them into 0600 instead of
 relying on the user's umask.
 .
 This change prevents hostnames that are visited while using sshuttle from
 being unintentionally leaked in this file when using -H.
 .
 The cache file is removed on exit, but it might still be kept in the
 system in case of abnormal program termination.
Author: Javier Fernández-Sanguino Peña <jfs@debian.org>
Bug-Debian: https://bugs.debian.org/760886
Forwarded: no
Last-Update: <2015-06-30>

--- a/src/firewall.py
+++ b/src/firewall.py
@@ -568,7 +568,7 @@
         os.chmod(tmpname, st.st_mode)
     else:
         os.chown(tmpname, 0, 0)
-        os.chmod(tmpname, 0644)
+        os.chmod(tmpname, 0600)
     os.rename(tmpname, HOSTSFILE)
 
 
--- a/src/hostwatch.py
+++ b/src/hostwatch.py
@@ -37,6 +37,7 @@
         for name, ip in sorted(hostnames.items()):
             f.write('%s,%s\n' % (name, ip))
         f.close()
+        os.chmod(tmpname, 0600)
         os.rename(tmpname, CACHEFILE)
     finally:
         try:
